clang/include/clang/AST/OperationKinds.def

```diff
@@ -367,6 +367,9 @@ CAST_OPERATION(HLSLVectorTruncation)
 // Non-decaying array RValue cast (HLSL only).
 CAST_OPERATION(HLSLArrayRValue)
 
+// Aggregate by Value cast (HLSL only).
+CAST_OPERATION(HLSLAggregateCast)
```

Comment:
nit: How about we call this `HLSLElementwiseCast`? Aggregate cast doesn't really describe how the cast is performed or what it is doing.

Reply:
What about HLSLFlatCast ?